resource/store.go

```diff
@@ -154,6 +157,53 @@ func (s *Store) UpdateSubresource(
 	})
 }
 
+// Upsert updates/creates the provided object.
+// Keep in mind that an Upsert will completely overwrite the object,
+// so nil or missing values will be removed, not ignored.
+// It is usually best to use the result of a Get call, change the appropriate values, and then call Update with that.
+// The update will fail if no ResourceVersion is provided, or if the ResourceVersion does not match the current one.
+// It returns the updated/created Object from the storage system.
+func (s *Store) Upsert(ctx context.Context, obj Object) (Object, error) {
+	if obj.StaticMetadata().Kind == "" {
+		return nil, fmt.Errorf("obj.StaticMetadata().Kind must not be empty")
+	}
+	if obj.StaticMetadata().Namespace == "" {
+		return nil, fmt.Errorf("obj.StaticMetadata().Namespace must not be empty")
+	}
+	if obj.StaticMetadata().Name == "" {
+		return nil, fmt.Errorf("obj.StaticMetadata().Name must not be empty")
+	}
+
+	client, err := s.getClient(obj.StaticMetadata().Kind)
+	if err != nil {
+		return nil, err
+	}
+
+	resp, err := client.Get(ctx, obj.StaticMetadata().Identifier())
+
+	if err != nil {
+		cast, ok := err.(*k8sErrors.ServerResponseError)
```

Comment:
This dovetails a bit into https://github.com/grafana/grafana-app-sdk/issues/19, but I think it may be a better approach to define an interface like
```go
type APIServerResponseError interface {
    error
    StatusCode() int
}
```
And try casting to that to check the error status code, which allows us to avoid the dependency cycle and not move around the error types in the `k8s` package. Thoughts?